### PR TITLE
Image caching improvements using Kingfisher

### LINKED ISSUE #1432: Image caching improvements
Now that we're implementing Product management, images become more visible on the Products tab and also in each Product (read-only and edit version). Right now, we can use `UIImageView+Networking` to load an image on an `UIImageView` and `WPImageSource` to download an image from a URL. However, they don't cache the images on disk so the placeholder image is shown on each cell reuse and table/collection view load.

There are popular libraries like [`SDWebImage`](https://github.com/SDWebImage/SDWebImage), but let's consider implementing a lightweight version that suits our usage first. A possible implementation could be like this, but it might have some missing features: https://medium.com/swift2go/swiftbits-caching-images-9de11d034924

### PR DESCRIPTION
**Pull request by pmusolino**
Fixes #1432 

## Summary
We decided to go for [Kingfisher](https://github.com/onevcat/Kingfisher), after evaluating together what is the most efficient choice at this time.
This PR implements Kingfisher, which substitutes [UIImageView+Networking](https://github.com/wordpress-mobile/WordPressUI-iOS/blob/develop/WordPressUI/Extensions/UIImageView%2BNetworking.swift) as "engine" for downloading and caching images. The main difference and benefit is the cache on disk, but in the future, we will probably start using other useful features like image manipulation, animations, and the loader.
I decided to use the default configuration, but feel free to suggest a configuration that better reflects our needs:
- Total memory cache is 25% of your total memory in the device, and there is no limit on image count.
- Images in memory storage will expire after 5 minutes from last accessed, while it is a week for images in disk storage.

## Changes
- Added `Kingfisher` pod
- Removed some unused import of `WordPressUI`
- Implemented Kingfisher in our extension `UIImageView+Kingfisher` and substituted all the download occurrences of `UIImageView+Networking`
- Fixed an issue on content mode when the placeholder is showed in `ProductImageCollectionViewCell`

## Testing
Check that anything continues to work as expected in all the points where we use downloaded images, like Orders, Product List, and Product Detail.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pmusolino** [comment]
I created a small wrapper around the main method of Kingfisher, we will be able to extend it when we need to use other features. Ready for a new review 😉

**Comment by pmusolino** [comment]
Updated also the PR description

File: WooCommerce/Classes/Extensions/UIImageView+Kingfisher.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    // Base method to download and cache images
+    //
+    func setImage(
+        with url: String?,
+        placeholder: UIImage? = nil,
+        progressBlock: ImageDownloadProgressBlock? = nil,
+        completionHandler: ImageDownloadResultBlock? = nil) {
+
+        let url = URL(string: url ?? "")
+        self.kf.setImage(with: url, placeholder: placeholder, progressBlock: progressBlock) { (result) in
```
Comment (jaclync) [inline WooCommerce/Classes/Extensions/UIImageView+Kingfisher.swift]: nit: is `self` needed?

File: WooCommerce/Classes/Extensions/UIImageView+Kingfisher.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import UIKit
+import Kingfisher
+
+/// Wrapper Extension for Kingfisher
+///
+extension UIImageView {
+
+    public typealias ImageDownloadProgressBlock = ((_ receivedSize: Int64, _ totalSize: Int64) -> Void)
+    public typealias ImageDownloadResultBlock = ((_ success: Bool) -> Void)
```
Comment (jaclync) [inline WooCommerce/Classes/Extensions/UIImageView+Kingfisher.swift]: nit: could these two stay internal instead of public?

**Pull request merged**